model/src/main/resources/reference.yml

```diff
@@ -40,6 +40,7 @@ ort:
   # configurations for the same package, see https://github.com/oss-review-toolkit/ort/issues/6972 for details.
   packageConfigurationProviders:
   - type: DefaultDir
+  - type: OrtConfig
```

Comment:
Unrelated to this specific PR, but it just struck me that in line 44 below we have `type: Dir`, whereas for `packageCurationProviders` we have `type: File` even if it's in fact a directory. I believe we should align that.

And I also just realized that we have `DirectoryPackageConfigurationProvider` *and* `DirPackageConfigurationProvider`, both inheriting from `SimplePackageConfigurationProvider`... huh!? I guess some clean-up is needed here.

Reply:
I just checked, and both `Dir(ectory)PackageConfigurationProvider` work when being passed just a single `File` instead of a directory.

Reply:
Allowing single files is an unintended functionality. It also should only work with single `PackageConfiguration`s in this file, as multiple package configurations in a single files are no longer supported since #6831.

So I think that this behavior should be fixed and only directories, instead of renaming `Dir` to `File`.

Reply:
> So I think that this behavior should be fixed and only directories, instead of renaming `Dir` to `File`.

I'm not yet convinced, as it could make sense to align with `PackageCurationProvider` instead.

Anyway, this is all not strictly related to this PR and could be addressed independently.

Reply:
If we do a renaming, the curation provider should be renamed from `File` to `Dir`, treating the single file input as a special case, because for package configurations we won't support single files again.
And I agree that we should not have both a `DirectoryPackageConfigurationProvider` and a `DirPackageConfigurationProvider`.

Reply:
> because for package configurations we won't support single files again.

Note that there are two distinct "single file things" that we should not mix up:

1. Storing a multiple package configuration in a single file. This is the case that is not supported anymore.
2. Pointing a `Dir(ectory)PackageConfigurationProvider` at a single file containing a single package configuration. This works coincidentally, and I do not see a reason to stop supporting that.

Reply:
Wouldn't it be weird to have a provider that can only provide a single package configuration for a single package?

Reply:
That's not what I'm suggesting to do. I'm just saying that *if* someone configures a `Dir(ectory)PackageConfigurationProvider` to point to a file instead of a directory, I'm fine if that "just works" (as it is now) instead of e.g. throwing an exception.

Reply:
But as it makes little sense to configure a provider for a single package configuration it is likely a configuration mistake in which case it would be more helpful for the user to throw an exception.